### DISCUSSION TITLE
x-mask real time formatter

### DISCUSSION THREAD

**Discussion by MaquinaTech** [alpinejs/alpine#4107] [7. Bugs (archived)] [answered]
When use x-mask directive, with :dynamic option, and delete numbers after you write more than 3 numbers the format of input not found.

Example: input x-model="total" x-mask:dynamic="$money($input, ',', '.')"
                        type="text" id="total" name="total"

When write 1.222 and delete the last number, the result is 1.22

**Comment by ekwoka** [comment]
What should it do?

I believe the logic exists that it mostly avoids formatting on backspaces or you could keep adding more and more things the user is trying to delete.

But I'm not sure what you expect to happen when someone presses backspace there.

**Reply by MaquinaTech** [reply]
I think when you press the backspace key it should change the number format, if you type 2000 and format the x mask to 2.000 and then delete the last number. It should change the format, currently it is left like this: 2.00 but you should change it to 200 because it may cause confusion in the formats

**Reply by SimoTod** [reply]
I think there's a language misunderstanding. The period is a decimal separator in english speaking countries. I assume it's a thousands separator in your case (hence the bug report), right?

**Reply by SimoTod** [reply, marked as answer]
This is caused by https://github.com/alpinejs/alpine/blob/main/packages/mask/src/index.js#L71
I'm not sure what are the side effects of removing it. Maybe something to do with the cursor or maybe it was just an attempt to over optimising the plugin since it's been there since day 1. 🤷

**Reply by MaquinaTech** [reply]
Yes, I try to separate thousands with '.' and decimals with ','

**Reply by ekwoka** [reply]
It's generally required, not over optimized.

Since a mask for dates of `99/99/9999` will inject the `/` after you type `12`. Then the user can backspace to remove the `/` to edit the first number, without that, the user could never delete the `/` or even remove the decimal separator in money numbers. So `2.000,00` if the user hits backspace, how should it handle deleting the `,` when the user gets to it?

I think it could be maybe worked around...like don't reinject injected characters, but reformat what is already there...but it would take a bit more...

Maybe your use case would prefer the lazy mask instead
